githubscript/src/githubscript/release_config.py

```diff
@@ -18,6 +18,14 @@ def get_release_config(product_config, task_payload, config):
 
         owner = task_payload["githubOwner"]
         repo_name = task_payload["githubRepoName"]
+        config_owner = product_config.get("github_owner", "")
+        config_repo = product_config.get("github_repo_name", "")
```

Comment:
I think these will be `""` since we don't set the owner and repo in `allow_github_repo_override` repos: https://github.com/mozilla-releng/scriptworker-scripts/blob/master/githubscript/docker.d/worker.yml#L45-L49.

This check should instead be something like

```
expected_project_prefix = product_config["scope_project"].rstrip("*")
if not f"{owner}/{repo_name}".startswith(expected_project_prefix):
    raise TaskVerificationError(...)
```

Essentially, `product_config["scope_project"]` will be something like `mozilla-extensions/*` or `mozilla-releng/staging-xpi-*`. `f"{owner}/{repo_name}"` will be something like `mozilla-releng/staging-xpi-public` or `mozilla-extensions/normandy-devtools` or `mozilla-mobile/fenix` if someone's trying to do something sneaky. We need to make sure that the repo we're pointing at from the task definition starts with the `scope_product` minus the final `*`.

If that makes sense to you and @bhearsum, and you can get this fixed up and landed when I'm out, great. If this needs to wait til I get back, that's ok.